indico/modules/events/timetable/controllers/legacy.py

```diff
@@ -315,18 +317,23 @@ def _process(self):
         entries = []
         day = dateutil.parser.parse(data['day']).date()
         query = Contribution.query.with_parent(self.event_new).filter(Contribution.id.in_(data['contribution_ids']))
-        for contribution in query:
-            start_dt = find_next_start_dt(contribution.duration,
-                                          obj=self.session_block or self.event_new,
-                                          day=None if self.session_block else day)
-            # TODO: handle scheduling not-fitting contributions
-            #       can only happen within session blocks that are shorter than contribution's duration
-            if start_dt:
-                entries.append(self._schedule(contribution, start_dt))
-        return jsonify(entries=[serialize_entry_update(x) for x in entries])
+        with track_time_changes(auto_extend='end', user=session.user) as changes:
+            for contribution in query:
+                start_dt = find_next_start_dt(contribution.duration,
+                                              obj=self.session_block or self.event_new,
+                                              day=None if self.session_block else day,
+                                              force=True)
+                entry = self._schedule(contribution, start_dt)
+                if entry.end_dt.date() > day:
```

Comment:
We should really add at some point `start_dt_local`/`end_dt_local` autogenerated with a mixin.